dt-assets/js/service-worker.js

```diff
@@ -0,0 +1,139 @@
+self.addEventListener("install", () => {
+  self.skipWaiting();
+});
+
+self.addEventListener('notificationclick', (event) => {
+  const linkURL = event.notification.data.link;
+
+  event.notification.close();
+
+  event.waitUntil(
+    clients.matchAll({ type: 'window' }).then((clientList) => {
+      for (const client of clientList) {
+        if (client.url === '/' && 'focus' in client) {
+          return client.focus();
+        }
+      }
+      if (clients.openWindow && linkURL) {
+        return clients.openWindow(linkURL);
+      }
+    })
+  );
+});
+
+addEventListener("message", (event) => {
+
+  const url = '/wp-json/dt/v1/notifications/get_notifications';
+  let payload = {
+    all: false,
+    page: 0,
+    limit: 100,
+    mentions: false,
+  }
+
+  let last_checked = new Date();
+  event.waitUntil(setInterval(function() {
+    fetch(url,{
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json; charset=utf-8",
+        "X-WP-Nonce": event.data["nonce"],
+      },
+      body: JSON.stringify(payload),
+    })
+      .then((response) => {
+        return response.json();
+      })
+      .then((json) => {
+        
+        for (let i = 0; i < json.length; i++) {
```

Comment:
Instead of a basic for loop, you can use for..of to get each item as well as the index:

```
for (const [index, notification] of notifications.entries()) {
  ...
}
```